### PR TITLE
Disable useUserOverride in CultureContext

### PR DESCRIPTION
**Pull request by khyperia**
This is a test-only change.

[CultureInfo has this documentation](https://msdn.microsoft.com/en-us/library/ky2chs3h(v=vs.110).aspx): The user might choose to override some of the values associated with the current culture of Windows through the regional and language options portion of Control Panel. For example, the user might choose to display the date in a different format or to use a currency other than the default for the culture. **If the culture identifier associated with name matches the culture identifier of the current Windows culture, this constructor creates a CultureInfo object that uses those overrides**, including user settings for the properties of the DateTimeFormatInfo instance returned by the DateTimeFormat property, and the properties of the NumberFormatInfo instance returned by the NumberFormat property.

This was causing one test on my machine to fail:

```
Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.UnitTests.ExpansionTests.Primitives

Assert.Equal() Failure\r\n           ↓ (pos 1)\r\nExpected: {1/1/0001 12:00:00 AM}\r\nActual:   {0001-01-01 00:00:00}\r\n           ↑ (pos 1)

   at Microsoft.CodeAnalysis.ExpressionEvaluator.ResultProviderTestBase.Verify(DkmEvaluationResult actual, DkmEvaluationResult expected)
   at Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.UnitTests.ExpansionTests.Primitives() in C:\c\roslyn\src\ExpressionEvaluator\CSharp\Test\ResultProvider\ExpansionTests.cs:line 62
```

The fix is to explicitly override this behavior, by passing `useUserOverride = false` in a different constructor.

Behavior testing in a console app:

```
Console.WriteLine(new DateTime());
CultureInfo.CurrentCulture = new CultureInfo("en-US");
Console.WriteLine(new DateTime());
CultureInfo.CurrentCulture = new CultureInfo("en-US", false);
Console.WriteLine(new DateTime());
```

produces

```
0001-01-01 00:00:00
0001-01-01 00:00:00
1/1/0001 12:00:00 AM
```

---

I don't know what team should review this, so I've just added roslyn-compiler. Feel free to add a `area-*` and the correct reviewer if there is a more appropriate team.

### CODE REVIEW

**Comment by khyperia** [comment]
... _never mind_, apparently [that constructor](https://apisof.net/catalog/System.Globalization.CultureInfo..ctor(String,Boolean)) is not available in `netstandard1.3`, which is what we are compiling for in [`TestUtilities.csproj`](https://github.com/dotnet/roslyn/blob/master/src/Test/Utilities/Portable/TestUtilities.csproj#L13).

😢

File: src/Compilers/CSharp/Test/Emit/PDB/PDBConstantTests.cs [hunk 1]
```diff
@@ -476,7 +476,7 @@ void M()
     }
 }
 ";
-            using (new CultureContext("en-US"))
+            using (new CultureContext(new CultureInfo("en-US", false)))
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBConstantTests.cs]: >false [](start = 63, length = 5)

Consider using named argument.

**Comment by TyOverby** [comment]
LGTM

**Comment by khyperia** [comment]
Pushed another commit that both takes into account @AlekseyTs's feedback to use named arguments, as well as doing a regex search for all `CultureInfo` constructors and fixing those too (but only in tests - there are a couple uses in non-test code, but I left those alone)

**Comment by khyperia** [comment]
retest windows_release_vs-integration_prtest please

**Pull request merged**